### ISSUE TITLE
Give an example line of code 

### ISSUE THREAD

**Issue by Redirect0null** [taras42/report_telegram_channels#4]
Give an example line of code  for a Python and Docker please 

Is this right line 
`python3 report.py <api_id> <api_hash> "Terrorism" https://t.me/siloviki_chat https://t.me/siloviki_belarus_chat https://t.me/Aksenov82 https://t.me/chp_crimea https://t.me/poptatiana http://t.me/Radioskaner_chat @Galandec_ZP  https://t.me/agentgosdepa1

`

**Comment by taras42** [comment]
I tested it with '@' only, not sure if links to t.me are working, the api is a bit vague about that. If it returns True in the output for t.me links that means it should work.

There's an example in the README, but it's generic and does not mention a specific channel:

```
python3 report.py <api_id> <api_hash> "<report_message>" @channel1 @channel2 @channel3
```

**Comment by taras42** [comment]
Also, you should not mention your api_id and api_hash on the internet. Someone might use it against you. I edited your comment, so it no loner mention your credentials.

**Comment by Redirect0null** [comment]
@taras42  Thanks! 
`python3 report.py <api_id> <api_hash> "Terrorism" @Galandec_ZP `- must be right line of code ?

**Comment by taras42** [comment]
Yes. I updated the README with a better example.

**Comment by Redirect0null** [comment]
@taras42  ok 
But I do something wrong ( 
I have to install more tools for a Windows machine 
=== 
Щось не так ( 
Схоже, у мені потрібно поставити ще якісь тулзи 

![image](https://user-images.githubusercontent.com/100712266/156225889-62d28f87-cdf0-4ddd-87a6-2e2ca1112b3d.png)

**Comment by taras42** [comment]
1. Відкриваєте cmd.
2. pip3 install telethon
3. Тут треба зайти в папку де зберегли report.py (мабуть треба додати це в документацію). Не пам'ятаю як точно на віндовсі, можливо вам прийдеться погуглити. (докер в цьому плані зручніший, але на віндовсі з ним також можуть бути проблеми)
5. Потім можна запускати python3 report.py <api_id> <api_hash> "Terrorism" @Galandec_ZP

Нажаль віндовс не найкраща операційка для таких цілей. Буду дивитися що там з докером, це зручніше ніж встановлювати залежності руками.

**Comment by sprilukin** [comment]
Може треба якось через virtualenv зробити? не буде простіше?

**Comment by taras42** [comment]
ага

**Comment by Redirect0null** [comment]
MAC 11.2.1  
Library/Frameworks/Python.framework/Versions/3.10/bin/python3: can't open file '/Users/UserName/report.py': [Errno 2] No such file or directory

**Comment by sprilukin** [comment]
how are you running this command? please try run python in a folder which contains *.py file like
`python3 ./report.py .....`

**Comment by sprilukin** [comment]
also i would consider using Docker - it's much easier to use it than trying to fix all mythical python errors (i'm not a python dev😀)

**Comment by Redirect0null** [comment]
@sprilukin  Чесно кажучи з докером також не все просто (на вінді), а на макові, ще не пробував. 
На приклад на макові, я з нуля встановив Пітон, то в якій папці потрібно запустити консольку? 
Bush я зрозумію.

**Comment by Redirect0null** [comment]
зараз на макові спробую встановити докер і затещу

**Comment by Redirect0null** [comment]
на MAC OS 
![image](https://user-images.githubusercontent.com/100712266/156258412-790be81a-8e29-4660-a8c8-360804142641.png)

**Comment by Redirect0null** [comment]
WINDOWS 
![image](https://user-images.githubusercontent.com/100712266/156259393-df646605-e847-4e15-9537-f7add167324f.png)

**Comment by taras42** [comment]
Виглядає добре, пробували просто запустити з командного рядка без докера:

```
python3 report.py <api_id> <api_hash> "Terrorism" @Galandec_ZP
```

(<api_id> замініти на ваш айді, <api_hash> замінити на ваш api_hash)

**Comment by Redirect0null** [comment]
@taras42  
Те, що треба? 
Відпрацювало, як очікувалося? 

![image](https://user-images.githubusercontent.com/100712266/156268400-60b53712-b346-49e7-b246-46efc75b8ba3.png)

**Comment by taras42** [comment]
```
@Galandec_ZP True
```

Ніби відпрацювало. Не мало би бути помилки з Event Loop, але це мабуть специфіка Windows. Потім подивлюся чи можна то зафіксити.

**Comment by Redirect0null** [comment]
Чи є подібні застосунки для тікТоку, інстаграму та інших ?

**Comment by taras42** [comment]
Не в курсі для інших. Для тіктоку немає АПІ, можна хіба скрипт написати котрий буде клікати куди треба. Тут є POC.
[poc.zip](https://github.com/taras42/report_telegram_channels/files/8171877/poc.zip)
.

1. Install nodejs, npm.
2. Unzip, open terminal, go to the unzip folder
3. Run npm install
4. USERS=@tiktokUserName1,@tiktokUserName2 node report.js to report users (doesn't work in Windows CMD, gitbash will do)

**Comment by Redirect0null** [comment]
хз, треба пробувати, бо тікТокерів і інстаблогерів, повно.

**Comment by Redirect0null** [comment]
@sprilukin  @taras42 @nazavr322 
Використовую на віндовс, через Python 
Скрипт не блокує або не бачить користувачів, ботів, що починаються з `@`

![image](https://user-images.githubusercontent.com/100712266/156575835-b398be46-7c8f-4c26-a5ec-5d1461df690c.png)

**Comment by Redirect0null** [comment]
В ТГ подібні канали, при отриманні в повідомленнях - не клікабельні 
https://t.me/rlz_the_kraken
https://t.me/theshard18
https://t.me/faktiru
![image](https://user-images.githubusercontent.com/100712266/156582180-9ed4e4b1-00d0-405b-a07b-36cdf94ea299.png)

**Comment by taras42** [comment]
@Redirect0null Деякі юзери можуть вже не існувати. Я дивлюся на попередньому скріншоті юзер з @ був заріпорчений успішно.
![156268400-60b53712-b346-49e7-b246-46efc75b8ba3](https://user-images.githubusercontent.com/1831258/156595296-84aa9025-bc4e-4bb1-865f-a532f939a2a6.png)

**Comment by Redirect0null** [comment]
![image](https://user-images.githubusercontent.com/100712266/156652486-61bfee76-b140-481a-aa31-7f84cb78705a.png)

**Comment by taras42** [comment]
Так, справді. Чекнув на лінуксі, все ок. Подивлюся що там на вінді і чому воно не робить. (ви копіювали лінк?, спробуйте набрати вручну, можливо при копіюванні вставляються якісь символи котрі теоеграм апі не розпізнає)

**Comment by Redirect0null** [comment]
Я скопіював лінк з пошуку в телеграм, а потім сюди. 
Зараз пробую дублювати `@` =>  `https://t.me/ ` 
Результату поки нема.

**Comment by Redirect0null** [comment]
не працює ( 
https://t.me/post11x_reborn @post11x_reborn https://t.me/yug_plazdarm @yug_plazdarm https://t.me/SolovievLive @SolovievLive https://t.me/sbagdasarov @sbagdasarov http://t.me/addstickers/Sbagdasarov https://t.me/oldhellboy @oldhellboy https://t.me/oldhellb0y @oldhellb0y https://t.me/novostirussia24 @novostirussia24 https://t.me/rossia_news @rossia_news https://t.me/vrogov @vrogov https://t.me/majorselivanov @majorselivanov @kristinaa94 @ujniiplatsdarm https://t.me/Russkyship https://t.me/boris_rozhin https://t.me/post11x_reborn



![image](https://user-images.githubusercontent.com/100712266/156654157-ca1516df-78bb-4678-95ac-e6d0b39837a8.png)

**Comment by Redirect0null** [comment]
А докер не запускається ((  Віндовс 10 
Помилка
![image](https://user-images.githubusercontent.com/100712266/156659083-ca8b6ca8-2773-457a-a305-5680d946ae44.png)

**Comment by taras42** [comment]
Я подивлюся вінду сьогоді.

Пробували білдити так?

```
DOCKER_BUILDKIT=0 docker build . -t report-telegram
```

**Comment by Redirect0null** [comment]
Не виходить
![image](https://user-images.githubusercontent.com/100712266/156659852-89c0225f-0f56-49bc-bc2b-a252547c76ff.png)
![image](https://user-images.githubusercontent.com/100712266/156660021-b1190a3e-21f5-4b7d-9b46-6dca6bc26f73.png)

**Comment by taras42** [comment]
Ну хоча б збілдило вже, додам в README.

**Comment by Redirect0null** [comment]
так, але проблему не вірішена ні докером, ні Пітоном. 
Подивіться будь ласка, коли матимете час. 
Ви сильно допоможете цим. 
Дякую!

**Comment by taras42** [comment]
Я не зміг зарепродюсити в себе, в мене воно знаходить канали. Пофіксив помилку на віндовс коли скрипт завершує виконання і тепер воно буде виводити справжню помилку, а не generic no user .... Тому обновіть скрипт, запустіть ще раз і тоді воно має вивести реальну помилку, можливо є якась проблема локально.

**Comment by taras42** [comment]
Якшо так і не буде локально працювати, на вінді можна спробувати це: https://docs.microsoft.com/en-us/windows/wsl/install. Воно поставить ядро лінукса і тоді проблем не має бути. (можна буде використовувати лінуксовський термінал у віндовсі)
